ci/infra/testrunner/tests/conftest.py

```diff
@@ -38,19 +39,33 @@ def cleanup():
 
 @pytest.fixture
 def bootstrap(request, provision, skuba):
+
     if request.config.getoption("skip_setup") in ['bootstrapped', 'deployed']:
         return
 
     skuba.cluster_init()
     skuba.node_bootstrap()
 
 
+def check_system_pods_ready(kubectl):
+    pods = json.loads(kubectl.run_kubectl('get pods --namespace=kube-system -o json'))['items']
+    for pod in pods:
+        pod_status = pod['status']['phase']
+        pod_name   = pod['metadata']['name']
+        assert pod_status in ['Running', 'Completed'], f'Pod {pod_name} status {pod_status} != Running or Completed'
+
 @pytest.fixture
 def deployment(request, bootstrap, skuba, kubectl):
     if request.config.getoption("skip_setup") != 'deployed':
         skuba.join_nodes()
 
-    wait(kubectl.run_kubectl, 'wait --timeout=5m --for=condition=Ready pods --all --namespace=kube-system', wait_delay=60, wait_timeout=300, wait_backoff=30, wait_retries=5, wait_allow=(RuntimeError))
+    wait(check_system_pods_ready,
+         kubectl,
+         wait_delay=60,
+         wait_timeout=10,
```

Comment:
I would prefer if we did not change the `wait_timeout` this pass as that is not what this PR is about. same with `wait_backoff` `wait_retires` and `wait_allow` it just seems like these changes are not related.